File: app/src/main/java/com/sabihamumcu/kelimekutusu/repo/localData/WordDao.java
package com.sabihamumcu.kelimekutusu.repo.localData;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.sabihamumcu.kelimekutusu.model.Word;

import java.util.List;

@Dao
public interface WordDao {

    @Insert
    void insert(Word word);

    @Update
    void update(Word word);

    @Delete
    void delete(Word word);

    @Query("Select * FROM word")
    LiveData<List<Word>> loadAll();

    @Query("Select word FROM word where id =:id")
    String getWordTitle(String id);

    @Query("Select * FROM word where id =:id")
    LiveData<Word> getWord(String id);

    @Query("Select * from word where word=:word")
    Word get(String word);

}

File: app/src/main/java/com/sabihamumcu/kelimekutusu/repo/localData/WordDao.kt
package com.sabihamumcu.kelimekutusu.repo.localData

import androidx.lifecycle.LiveData
import androidx.room.*
import com.sabihamumcu.kelimekutusu.model.Word

@Dao
interface WordDao {
    @Insert
    fun insert(word: Word?)

    @Update
    fun update(word: Word?)

    @Delete
    fun delete(word: Word?)

    @Query("Select * FROM word")
    fun loadAll(): LiveData<List<Word?>?>?

    @Query("Select word FROM word where id =:id")
    fun getWordTitle(id: String?): String?

    @Query("Select * FROM word where id =:id")
    fun getWord(id: String?): LiveData<Word?>?

    @Query("Select * from word where word=:word")
    operator fun get(word: String?): Word?
}